cpp/src/hdbscan/detail/reachability.cuh

```diff
@@ -67,10 +68,71 @@ void core_distances(
   auto indices = thrust::make_counting_iterator<value_idx>(0);
 
   thrust::transform(exec_policy, indices, indices + n, out, [=] __device__(value_idx row) {
-    return knn_dists[row * min_samples + (min_samples - 1)];
+    return knn_dists[row * n_neighbors + (min_samples - 1)];
   });
 }
 
+/**
+ * Wraps the brute force knn API, to be used for both training and prediction
+ * @tparam value_idx data type for integrals
+ * @tparam value_t data type for distance
+ * @param[in] handle raft handle for resource reuse
+ * @param[in] X input data points (size m * n)
+ * @param[out] inds nearest neighbor indices (size n_search_items * k)
+ * @param[out] dists nearest neighbor distances (size n_search_items * k)
+ * @param[in] m number of rows in X
+ * @param[in] n number of columns in X
+ * @param[in] search_items array of items to search of dimensionality D (size n_search_items * n)
+ * @param[in] n_search_items number of rows in search_items
+ * @param[in] k number of nearest neighbors
+ * @param[in] metric distance metric to use
+ */
+template <typename value_idx, typename value_t>
+void compute_knn(const raft::handle_t& handle,
+                 const value_t* X,
+                 value_idx* inds,
+                 value_t* dists,
+                 size_t m,
+                 size_t n,
+                 const value_t* search_items,
+                 size_t n_search_items,
+                 int k,
+                 raft::distance::DistanceType metric)
+{
+  auto stream      = handle.get_stream();
+  auto exec_policy = handle.get_thrust_policy();
+  std::vector<value_t*> inputs;
+  inputs.push_back(const_cast<value_t*>(X));
+
+  std::vector<int> sizes;
+  sizes.push_back(m);
+
+  // This is temporary. Once faiss is updated, we should be able to
+  // pass value_idx through to knn.
+  rmm::device_uvector<int64_t> int64_indices(k * n_search_items, stream);
```

Comment:
I *think* we might no longer need to do this. Can you try removing this and have it compute trhe output indices directly? I know we no longer need this in the impl detail API in RAFT and I think we *shouldn't* need it in the public API anymore either.